<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Cool Beans • CoolBeans</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Cool Beans">
<meta property="og:description" content="CoolBeans">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CoolBeans</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Johnson_R_Package_Tutorial.html">Cool Beans</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Cool Beans</h1>
                        <h4 data-toc-skip class="author">Soren Johnson</h4>
            
            <h4 data-toc-skip class="date">12/2/2021</h4>
      
      
      <div class="hidden name"><code>Johnson_R_Package_Tutorial.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Load in libaries used by this package</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ── <span style="font-weight: bold;">Attaching packages</span> ─────────────────────────────────────── tidyverse 1.3.1 ──</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BB00;">✓</span> <span style="color: #0000BB;">tibble </span> 3.1.6     <span style="color: #00BB00;">✓</span> <span style="color: #0000BB;">dplyr  </span> 1.0.7</span>
<span class="co">## <span style="color: #00BB00;">✓</span> <span style="color: #0000BB;">tidyr  </span> 1.1.4     <span style="color: #00BB00;">✓</span> <span style="color: #0000BB;">stringr</span> 1.4.0</span>
<span class="co">## <span style="color: #00BB00;">✓</span> <span style="color: #0000BB;">readr  </span> 2.1.1     <span style="color: #00BB00;">✓</span> <span style="color: #0000BB;">forcats</span> 0.5.1</span>
<span class="co">## <span style="color: #00BB00;">✓</span> <span style="color: #0000BB;">purrr  </span> 0.3.4</span></code></pre>
<pre><code><span class="co">## ── <span style="font-weight: bold;">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──</span>
<span class="co">## <span style="color: #BB0000;">x</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">filter()</span> masks <span style="color: #0000BB;">stats</span>::filter()</span>
<span class="co">## <span style="color: #BB0000;">x</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">lag()</span>    masks <span style="color: #0000BB;">stats</span>::lag()</span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Loading required package: usethis</span></code></pre>
<p>Install these packages and load them</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"vcfR"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"ape"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"poppr"</span><span class="op">)</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"bcm-uga/LEA"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/knausb/vcfR" class="external-link">vcfR</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##    *****       ***   vcfR   ***       *****</span>
<span class="co">##    This is vcfR 1.12.0 </span>
<span class="co">##      browseVignettes('vcfR') # Documentation</span>
<span class="co">##      citation('vcfR') # Citation</span>
<span class="co">##    *****       *****      *****       *****</span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/" class="external-link">ape</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://grunwaldlab.github.io/poppr/" class="external-link">poppr</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Loading required package: adegenet</span></code></pre>
<pre><code><span class="co">## Loading required package: ade4</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">##    /// adegenet 2.1.5 is loaded ////////////</span>
<span class="co">## </span>
<span class="co">##    &gt; overview: '?adegenet'</span>
<span class="co">##    &gt; tutorials/doc/questions: 'adegenetWeb()' </span>
<span class="co">##    &gt; bug reports/feature requests: adegenetIssues()</span></code></pre>
<pre><code><span class="co">## Registered S3 method overwritten by 'pegas':</span>
<span class="co">##   method      from</span>
<span class="co">##   print.amova ade4</span></code></pre>
<pre><code><span class="co">## This is poppr version 2.9.3. To get started, type package?poppr</span>
<span class="co">## OMP parallel support: available</span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://membres-timc.imag.fr/Olivier.Francois/lea.html" class="external-link">LEA</a></span><span class="op">)</span></code></pre></div>
<p>Now install own R package</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"sorenDJ/R_package_Johnson"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Skipping install of 'CoolBeans' from a github remote, the SHA1 (114ce99a) has not changed since last install.</span>
<span class="co">##   Use `force = TRUE` to force installation</span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"CoolBeans"</span><span class="op">)</span></code></pre></div>
<p>Let us download some data for our package:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/sorenDJ/R_package_Johnson/master/data/DNA_Samples_Data_Analysis.csv"</span>, destfile <span class="op">=</span> <span class="st">"/cloud/project/data/DNA_Samples_Data_Analysis.csv"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/sorenDJ/R_package_Johnson/master/data/mtDNA_samples_data.txt"</span>, destfile <span class="op">=</span> <span class="st">"/cloud/project/data/mtDNA_samples_data.txt"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/sorenDJ/R_package_Johnson/master/data/mtDNA_pop.vcf"</span>, destfile <span class="op">=</span> <span class="st">"/cloud/project/data/mtDNA_pop.vcf"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="working-example">Working Example<a class="anchor" aria-label="anchor" href="#working-example"></a>
</h2>
<ul>
<li>An example of each function and how it works</li>
<li>Say something about why this is a good thing to do</li>
<li>say something about expected output</li>
</ul>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mult.html">mult</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">5</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pop_data_output-function">pop_data_output function<a class="anchor" aria-label="anchor" href="#pop_data_output-function"></a>
</h2>
<ul>
<li><p>This function takes a csv file and converts it to a text file that only includes the data and samples that are specified.</p></li>
<li><p>This function is important because some population structure analyses require text files with population data and the samples you want to look at.</p></li>
</ul>
<p>Example:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">test_pop_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pop_data_output.html">pop_data_output</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/cloud/project/data/DNA_Samples_Data_Analysis.csv"</span>, columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">SJ_number</span>, <span class="va">Species</span>, <span class="va">sub_species</span><span class="op">)</span>, sample_col <span class="op">=</span> <span class="va">SJ_number</span>, samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AR01"</span>, <span class="st">"AR02"</span>, <span class="st">"AR03"</span><span class="op">)</span>, NA_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">""</span>, <span class="st">"na"</span><span class="op">)</span>, output_file <span class="op">=</span> <span class="st">"/cloud/project/test_pop_data.txt"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">265</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">3</span></span></code></pre>
<pre><code><span class="co">## <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">## <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="co">## <span style="color: #BB0000;">chr</span> (3): SJ_number, Species, sub_species</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## <span style="color: #00BBBB;">ℹ</span> Use <span style="color: #262626; background-color: #DADADA;">`spec()`</span> to retrieve the full column specification for this data.</span>
<span class="co">## <span style="color: #00BBBB;">ℹ</span> Specify the column types or set <span style="color: #262626; background-color: #DADADA;">`show_col_types = FALSE`</span> to quiet this message.</span></code></pre>
</div>
<div class="section level2">
<h2 id="samp_2b_seq-function">samp_2B_seq function<a class="anchor" aria-label="anchor" href="#samp_2b_seq-function"></a>
</h2>
<ul>
<li><p>This function is to be used to check the number of samples for different values of a variable and then filter out samples that have a certain value for the variable. These filtered samples will then be created as a new csv file.</p></li>
<li><p>This function can be helpful for organization of data. Often times I might want to know how many samples have already been sequeneced and how many haven’t. I then want a CSV file for the samples that haven’t been sequenced to know which ones they are.</p></li>
</ul>
<p>Example:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/samp_2B_seq.html">samp_2B_seq</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/cloud/project/data/DNA_Samples_Data_Analysis.csv"</span>, na_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">""</span>, <span class="st">"na"</span><span class="op">)</span>, col2count <span class="op">=</span> <span class="va">strange_topology</span>, output_csv <span class="op">=</span> <span class="st">"/cloud/project/test_output.csv"</span>, filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"yes"</span>, <span class="cn">NA</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">265</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">30</span></span></code></pre>
<pre><code><span class="co">## <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">## <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="co">## <span style="color: #BB0000;">chr</span> (25): SJ_number, Species, sub_species, race, Kris_clade, my_clades, samp...</span>
<span class="co">## <span style="color: #00BB00;">dbl</span>  (5): cluster, lat, long, ddRAD plate, Qubit_con_1(ng/ml)</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## <span style="color: #00BBBB;">ℹ</span> Use <span style="color: #262626; background-color: #DADADA;">`spec()`</span> to retrieve the full column specification for this data.</span>
<span class="co">## <span style="color: #00BBBB;">ℹ</span> Specify the column types or set <span style="color: #262626; background-color: #DADADA;">`show_col_types = FALSE`</span> to quiet this message.</span></code></pre>
<pre><code><span class="co">## <span style="color: #949494;"># A tibble: 3 × 2</span></span>
<span class="co">##   strange_topology     n</span>
<span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;">1</span> no                 118</span>
<span class="co">## <span style="color: #BCBCBC;">2</span> yes                 12</span>
<span class="co">## <span style="color: #BCBCBC;">3</span> <span style="color: #BB0000;">NA</span>                 135</span></code></pre>
</div>
<div class="section level2">
<h2 id="op_k_snmf-function">op_K_snmf function<a class="anchor" aria-label="anchor" href="#op_k_snmf-function"></a>
</h2>
<ul>
<li><p>This function is to be used to determine the optimal K value given a vcf data set using a sparse non-negative matrix factorization algorithm. This function will output a summary of the cross-validation values for each K and also a plot of K and minimum cross-validation. This function also creates a snmf project object.</p></li>
<li><p>This function can help to determine the number of ancestral populations and distinct genetic clusters. The found K value can then be used to plot structure plots. The created snmf project can then also be used for doing analyses in the LEA package.</p></li>
</ul>
<p>Example:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">op_K_snmf</span><span class="op">(</span>vcf_file <span class="op">=</span> <span class="st">"/cloud/project/data/mtDNA_pop.vcf"</span>, K <span class="op">=</span> <span class="fl">4</span><span class="op">:</span><span class="fl">5</span>, new_name <span class="op">=</span> <span class="st">"/cloud/project/test_1.geno"</span>, repetitions <span class="op">=</span> <span class="fl">2</span>, project_name <span class="op">=</span> <span class="st">"snmf_test_1"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="snmf_structure_plot-function">snmf_structure_plot function<a class="anchor" aria-label="anchor" href="#snmf_structure_plot-function"></a>
</h2>
<ul>
<li><p>This function creates a structure plot from a snmf project for a given value of K. The structure plot displays the probability of each sample belonging to a certain genetic cluster.</p></li>
<li><p>This function is important because strucutre plots are a common anlaysis a figure used to look at genetic structure.</p></li>
</ul>
<p>Example:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/snmf_structure_plot.html">snmf_structure_plot</a></span><span class="op">(</span>project_name <span class="op">=</span> <span class="va">snmf_test_1</span>, K <span class="op">=</span> <span class="fl">4</span>, colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"green"</span>, <span class="st">"blue"</span>, <span class="st">"purple"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="genlight_creator">genlight_creator<a class="anchor" aria-label="anchor" href="#genlight_creator"></a>
</h2>
<ul>
<li><p>This function creates a genlight object from a vcf file. This genlight object can then be used in the adegent pacakge to look at genetic population structure. The genlight object also combines a text file with the vcf data and one variable is chosen to be a population variable.</p></li>
<li><p>This function is important because it helps complete many steps in just one step. The genlight object is also important if using the adegent package and this function helps to keep the code organized so less likely to make a mistake.</p></li>
</ul>
<p>Example:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/genlight_creator.html">genlight_creator</a></span><span class="op">(</span>vcf_file <span class="op">=</span> <span class="st">"/cloud/project/data/mtDNA_pop.vcf"</span>, pop_data <span class="op">=</span> <span class="st">"/cloud/project/data/mtDNA_samples_data.txt"</span>, genlight_name <span class="op">=</span> <span class="st">"gl.test"</span>, pop_col <span class="op">=</span> <span class="va">sub_species</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 10
##   header_line: 11
##   variant count: 72
##   column count: 131
## 
Meta line 10 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 72
##   Character matrix gt cols: 131
##   skip: 0
##   nrows: 72
##   row_num: 0
## 
Processed variant: 72
## All variants processed</code></pre>
<pre><code><span class="co">## Warning in vcfR2genlight(new_vcf_object.VCF): Found 10 loci with more than two alleles.</span>
<span class="co">## Objects of class genlight only support loci with two alleles.</span>
<span class="co">## 10 loci will be omitted from the genlight object.</span></code></pre>
<pre><code><span class="co">## [1] "gl.test"</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Soren Johnson.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
